### PR TITLE
Enable Table Function and PromQL function

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
Enable Table function and promQL function. This also enables to build spark sql function.
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1019
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by rupal-bq — DISMISSED** [review]

**Comment by penghuo** [comment]
update the doc? https://github.com/opensearch-project/sql/blob/main/docs/user/ppl/admin/prometheus_connector.rst#ppl-query-support-for-prometheus-connector

**Comment by dai-chen** [comment]
I forgot why we disabled this in PPL earlier?

**Comment by vamsi-amazon** [comment]
> I forgot why we disabled this in PPL earlier?

@dai-chen  There was no UX on dashboards for table functions during 2.4 release. With spark sql we are building UX to handle table functions.

File: ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java [hunk 1]
```diff
@@ -357,10 +363,16 @@ public UnresolvedPlan visitTableSourceClause(TableSourceClauseContext ctx) {
...
-  @Generated //To exclude from jacoco..will remove https://github.com/opensearch-project/sql/issues/1019
   public UnresolvedPlan visitTableFunction(OpenSearchPPLParser.TableFunctionContext ctx) {
-    //<TODO>
-    return null;
+    ImmutableList.Builder<UnresolvedExpression> builder = ImmutableList.builder();
+    ctx.functionArgs().functionArg().forEach(arg
+        -> {
+      String argName = (arg.ident() != null) ? arg.ident().getText() : null;
+      builder.add(
+          new UnresolvedArgument(argName,
```
Comment (Yury-Fridlyand) [inline ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java:372]: When `argName` could be `null`? Are you protected from NPE when it is used?

File: ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java [hunk 2]
```diff
@@ -357,10 +363,16 @@ public UnresolvedPlan visitTableSourceClause(TableSourceClauseContext ctx) {
   }
 
   @Override
-  @Generated //To exclude from jacoco..will remove https://github.com/opensearch-project/sql/issues/1019
   public UnresolvedPlan visitTableFunction(OpenSearchPPLParser.TableFunctionContext ctx) {
```
Comment (Yury-Fridlyand) [inline ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java:366]: Maybe: move to import
```suggestion
  public UnresolvedPlan visitTableFunction(TableFunctionContext ctx) {
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by penghuo — APPROVED** [review]

File: ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java (hunk 2 above)
Comment (vamsi-amazon) [reply ppl/src/main/java/org/opensearch/sql/ppl/parser/AstBuilder.java:366]: done.

**Review by ps48 — APPROVED** [review]

**Review by anirudha — APPROVED** [review]
Awesome! @penghuo to signoff

**Review by penghuo — APPROVED** [review]

**Pull request merged**